src/screens/Exchange/HotSwapsList.js

```diff
@@ -0,0 +1,178 @@
+  border-bottom-width: 1px;
+`;
+
+const IconWrapper = styled.View`
+  align-items: center;
+  margin: 0 ${spacing.rhythm}px;
+`;
+
+const PopularSwapsText = styled(BaseText)`
+  color: ${themedColors.popularSwaps};
+  text-align: center;
+  ${fontStyles.tiny}
+`;
+
+const Pill = styled.TouchableOpacity`
+  background-color: ${({ placeholder }) => placeholder ? 'transparent' : themedColors.tertiary};
+  border-radius: 6px;
+  padding: 10px ${({ grid }) => grid ? 10 : 20}px;
+  align-items: center;
+  flex-direction: row;
+  justify-content: center;
+  flex: ${({ grid }) => grid ? 1 : 0};
+`;
+
+const SwapText = styled(BaseText)`
+  color: ${themedColors.primary};
+  ${fontStyles.small}
+`;
+
+const hotSwapsIcon = require('assets/icons/popular.png');
+
+const SwapPill = (props: SwapPillProps) => {
+  const {
+    from,
+    to,
+    grid = false,
+    onPress,
+    style,
+  } = props;
+
+  return (
+    <Pill onPress={onPress} grid={grid} style={style}>
+      <SwapText>{from} </SwapText>
+      <SwapText style={{ transform: [{ scaleX: -1 }] }}>
+        <Icon name="back" />
+      </SwapText>
+      <SwapText> {to}</SwapText>
+    </Pill>
+  );
+};
+
+export const HotSwapsHorizontalList = (props: Props) => {
+  const { onPress, swaps } = props;
+  return (
+    <WrappingContainer>
+      <FlatList
+        horizontal
+        data={swaps}
+        renderItem={
+          ({ item }) => (
+            <SwapPill
+              {...item}
+              onPress={() => onPress(item.from, item.to)}
+              style={{ marginHorizontal: 2 }}
+            />
+          )
+        }
+        keyExtractor={item => `${item.from}-${item.to}`}
+        ListHeaderComponent={
+          <IconWrapper>
+            <Image source={hotSwapsIcon} />
+            <PopularSwapsText>Popular{'\n'}swaps</PopularSwapsText>
+          </IconWrapper>
+        }
+        contentContainerStyle={{
+          alignItems: 'center',
+          paddingVertical: spacing.mediumLarge,
+          paddingRight: spacing.mediumLarge,
+        }}
+      />
+    </WrappingContainer>
+  );
+};
+
+export class HotSwapsGridList extends React.Component<Props> {
+  renderItem = ({ item }: {item: {from: string, to: string}}) => {
+    const { onPress } = this.props;
+    if (item.from) {
+      return (
+        <SwapPill
+          {...item}
+          onPress={() => onPress(item.from, item.to)}
+          grid
+          style={{ margin: spacing.mediumLarge / 2 }}
+        />
+      );
+    }
+    return (
+      <Pill
+        placeholder
+        grid
+        style={{ margin: spacing.mediumLarge / 2 }}
+      />
+    );
+  }
+
+  render() {
+    const { swaps } = this.props;
+
+    // on small devices 3 columns will cause text lines to be broken
+    // on larger devices 2 columns will cause pills to be too wide
+    const columns = screenWidth >= threeColumnsMinScreenWidth ? 3 : 2;
+    const data = [...swaps];
+    while (data.length % columns !== 0) {
+      data.push({});
+    }
```

Comment:
What about moving this to other lifecycle step rather than render? I. e. `componentDidUpdate` with check if data has changed?

Reply:
But why? It won't have much impact on performance imho.

Reply:
The problem is that it will iterate the while on each render while it's only needed for first time, right?

Reply:
Actually, it's a very quick operation, the cycle will iterate +-2 times only.
If render function itself will be called too many time, then it's gonna be another problem :)